### PR TITLE
[server][DBTables][LastInfo] Implement LastInfo2

### PR DESCRIPTION
**Pull request by cosmo0920**
Resend #1202.

### CODE REVIEW

File: server/src/DBTablesLastInfo.cc [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+}
+
+static string makeIdListCondition(const LastInfoIdList &idList)
+{
+	string condition;
+	const ColumnDef &colId = COLUMN_DEF_LAST_INFO[IDX_LAST_INFO_ID];
+	SeparatorInjector commaInjector(",");
+	condition = StringUtils::sprintf("%s in (", colId.columnName);
+	LastInfoIdListConstIterator it = idList.begin();
+	while (true) {
```
Comment (kz0817) [inline server/src/DBTablesLastInfo.cc]: This block can be written with few lines simply as
<pre>
for (; it != idList.end(); ++it) {
    const int &id = *it;
    commnadInjector(condition);
    condition += StringUtils::sprintf("%d", id); 
}

or

for (; it != idList.end(); ++it) {
    commnadInjector(condition);
    condition += StringUtils::sprintf("%d", *it);
}

</pre>

**Comment by kz0817** [comment]
Review had been done. I wrote one comment.

File: server/src/DBTablesLastInfo.cc (hunk 1 above)
Comment (cosmo0920) [inline server/src/DBTablesLastInfo.cc]: I like later one style.

But, we can use more sophisticated range-based for style:

```cpp
for (auto id : idList) {
	commaInjector(condition);
	condition += StringUtils::sprintf("%ld", id);
}
```

How about this one?

**Comment by cosmo0920** [comment]
I've fixed a reviewed issue.

File: server/src/DBTablesLastInfo.cc (hunk 1 above)
Comment (kz0817) [inline server/src/DBTablesLastInfo.cc]: Yes, I want to use *type* inference feature.

2015-05-19 11:42 GMT+09:00 Hiroshi Hatake <notifications@github.com>:

**Comment by kz0817** [comment]
I merge this PR.

BTW, recently the build on TravisCI often failed with internal error.
If the reason is the short of a memory, it is worth reducing the number of parallel build.

**Comment by cosmo0920** [comment]
Thanks!

**Pull request merged**